crates/configuration/src/global_settings.rs

```diff
@@ -54,49 +61,92 @@ impl Default for GlobalSettingsBuilder {
                 node_spawn_timeout: 300,
                 local_ip: None,
             },
+            errors: vec![],
         }
     }
 }
 
 impl GlobalSettingsBuilder {
-    pub fn new() -> GlobalSettingsBuilder {
+    pub fn new() -> Self {
         Self::default()
     }
 
-    fn transition(config: GlobalSettings) -> Self {
-        Self { config }
+    fn transition(config: GlobalSettings, errors: Vec<anyhow::Error>) -> Self {
+        Self { config, errors }
     }
 
-    pub fn with_bootnodes_addresses(self, addresses: Vec<MultiAddress>) -> Self {
-        Self::transition(GlobalSettings {
-            bootnodes_addresses: addresses,
-            ..self.config
-        })
+    pub fn with_bootnodes_addresses<T>(self, bootnodes_addresses: Vec<T>) -> Self
+    where
+        T: TryInto<Multiaddr> + ToString + Copy,
```

Comment:
Can we use Display here.

```suggestion
        T: TryInto<Multiaddr> + Display + Copy,
```